### ISSUE TITLE
`invalid schema:style:explode combination` when generating code

### ISSUE THREAD

**Issue by rkapka** [ogen-go/ogen#649]
I am trying to generate server code for https://github.com/ethereum/beacon-APIs. After fixing a few issues with missing enum types, I ran into the following mysterious error:
```
- beacon-node-oapi.yaml:8:7 -> invalid schema:style:explode combination: ("":"form":true)
            7 |         
        →   8 |     All requests by default send and receive JSON, and as such should have either or both of the "Content-Type: application/json"       
              |       ↑ 
            9 |     and "Accept: application/json" headers.  In addition, some requests can return data in the SSZ format.  To indicate that SSZ        
           10 |     data is required in response to a request the header "Accept: application/octet-stream" should be sent.  Note that only a subset    
           11 |     of requests can respond with data in SSZ format; these are noted in each individual request.        
```
I have no clue how I can fix this. Any pointers greatly appreciated

**Comment by tdakkota** [comment]
Validation fails here, because this schema does not have `type` property:
https://github.com/ethereum/beacon-APIs/blob/ff896567205d6256e1cc69b0cdcd007a9f61adda/apis/beacon/blocks/headers.yaml#L12-L14

 
Replace
```yaml
      schema:
        allOf:
          - $ref: "../../../beacon-node-oapi.yaml#/components/schemas/Uint64"
          - example: ""
```
with
```yaml
      schema:
        $ref: "../../../beacon-node-oapi.yaml#/components/schemas/Uint64"
```

and it would work fine.

**Comment by rkapka** [comment]
I added the `type` property, but more issues appeared. Unfortunately I could not solve them because the error pointed to an incorrect line. Thank you for opening #650, this is tremendously helpful. Unfortunately after updating to `v0.54` I get incorrect messages about missing files:
```
- beacon-node-oapi.yaml:235:13 -> get "home/radek/repo/eth2APIs/types/primitive.yaml": file: open home/radek/repo/eth2APIs/types/primitive.yaml: no such file or directory
          234 |     ForkVersion:        
        → 235 |       $ref: './types/primitive.yaml#/ForkVersion'       
              |             ↑   
          236 |     DependentRoot:      
          237 |       $ref: './types/primitive.yaml#/DependentRoot'     
          238 |     ExecutionOptimistic:
```

**Comment by tdakkota** [comment]
> Unfortunately after updating to v0.54 I get incorrect messages about missing files

This should be fixed by #657. Try the `main` version.

**Comment by tdakkota** [comment]
The [`v0.55.0`](https://github.com/ogen-go/ogen/releases/tag/v0.55.0) release includes fixes of the bugs noted here.

Command
```console
$ ogen --debug.noerr --infer-types --allow-remote https://raw.githubusercontent.com/ethereum/beacon-APIs/master/beacon-node-oapi.yaml
```
works fine.

So, I closing the issue.
